contracts/collect/StepwiseCollectModule.sol

```diff
@@ -0,0 +1,329 @@
+ * @param recipient The recipient address associated with this publication.
+ * @param referralFee The referral fee associated with this publication.
+ * @param followerOnly Whether only followers should be able to collect.
+ * @param endTimestamp The end timestamp after which collecting is impossible.
+ * @param a The a multiplier of x^2 in quadratic equation (how quadratic is the curve) (9 decimals)
+ * @param b The b multiplier of x in quadratic equation (if a==0, how steep is the line) (9 decimals)
+ * @param c The c constant in quadratic equation (aka start price) (18 decimals)
+ */
+struct StepwiseCollectModuleInitData {
+    uint64 collectLimit;
+    address currency;
+    address recipient;
+    uint16 referralFee;
+    bool followerOnly;
+    uint40 endTimestamp;
+    uint72 a;
+    uint56 b;
+    uint128 c;
+}
+
+/**
+ * @title StepwiseCollectModule
+ * @author Lens Protocol
+ *
+ * @notice This is a simple Lens CollectModule implementation, inheriting from the ICollectModule interface and
+ * the FeeCollectModuleBase abstract contract.
+ *
+ * This module works by allowing limited collects for a publication within the allotted time with a changing fee.
+ *
+ * The fee is calculated based on a simple quadratic equation:
+ *      Fee = a*x^2 + b*x + c,
+ *      (where x is how many collects were already performed)
+ *
+ * a=b=0 makes it a constant-fee collect
+ * a=0 makes it a linear-growing fee collect
+ */
+contract StepwiseCollectModule is FeeModuleBase, FollowValidationModuleBase, ICollectModule {
+    using SafeERC20 for IERC20;
+
+    // As there is hard storage optimisation of a,b,c parameters, the following decimals convention is assumed for fixed-point calculations:
+    uint256 public constant A_DECIMALS = 1e9; // leaves 30 bits for fractional part, 42 bits for integer part
+    uint256 public constant B_DECIMALS = 1e9; // leaves 30 bits for fractional part, 26 bits for integer part
+    // For C the decimals will be equal to currency decimals
+
+    mapping(uint256 => mapping(uint256 => ProfilePublicationData))
+        internal _dataByPublicationByProfile;
+
+    constructor(address hub, address moduleGlobals) FeeModuleBase(moduleGlobals) ModuleBase(hub) {}
+
+    /**
+     * @notice This collect module levies a fee on collects and supports referrals. Thus, we need to decode data.
+     *
+     * @param profileId The profile ID of the publication to initialize this module for.
+     * @param pubId The publication ID to initialize this module for.
+     * @param data The arbitrary data parameter, decoded into: StepwiseCollectModuleInitData struct
+     * @return bytes An abi encoded bytes parameter, containing a struct with module initialization data.
+     */
+    function initializePublicationCollectModule(
+        uint256 profileId,
+        uint256 pubId,
+        bytes calldata data
+    ) external override onlyHub returns (bytes memory) {
+        unchecked {
```

Comment:
Does this unchecked block do anything, or is it a relic?

Reply:
Yup, a relic :) Will remove